WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginListener.java

```diff
@@ -33,6 +35,8 @@
 
     // Login email password callbacks
     void needs2fa(String email, String password);
+    void needs2fa(String email, String password, String idToken, String service, boolean isSocialLogin);
+    void needs2fa(String email, String userId, String nonceAuthenticator, String nonceBackup, String nonceSms);
```

Comment:
There are multiple instances which two-factor authentication is required as you mentioned and each has a different set of parameters.  That seems like a good case to use overloaded methods.  I didn't rename the interface methods since one is strictly non-social, one is either social or not, and one is strictly social.

Since you can get to the two-factor login screen from the email/password screen two different ways (i.e. non-social login, social login with connect), the `isSocialLoginConnect` parameter is needed.

Reply:
`LoginListener` interface methods are deliberately descriptive in an attempt to address Login's complexity and correspond to the different flow or subflows so, let's follow suit here. Example: See `loggedInViaSignup(list)` vs `loggedInViaPassword(list)`. Those two could be defined overloaded but are not and instead, their names correspond to the login flow/subflow.

> I didn't rename the interface methods since one is strictly non-social, one is either social or not, and one is strictly social.
> 
> Since you can get to the two-factor login screen from the email/password screen two different ways (i.e. non-social login, social login with connect), the isSocialLoginConnect parameter is needed.

I might be missing something here so, in an attempt to understand the code better I prepared [this commit](https://github.com/wordpress-mobile/WordPress-Android/commit/c137578dbb97f30309dbd7bf31f43fe740223630) with the method names "de-overloaded". Can you have a look and check if it's missing any functionality? Thanks!

In particular, is there a need for that second overloaded version of `needs2fa` to be both for social and non-social?

Reply:
Yes, the [second version](https://github.com/wordpress-mobile/WordPress-Android/blob/545094c92ca1ec438b260c6d8e6a30cc5d072bc7/WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginListener.java#L38) of the `needs2fa` overloaded methods is needed for social and non-social instances currently.  It was designed that way so the [first version](https://github.com/wordpress-mobile/WordPress-Android/blob/545094c92ca1ec438b260c6d8e6a30cc5d072bc7/WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginListener.java#L37) could remain unaltered and files irrelevant to social login could also remained unaltered (i.e. [`LoginUsernamePasswordFragment`](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginUsernamePasswordFragment.java#L328)).

I compromised with 954b09fb6dcd83f699d0871081b6e9b2b927d3ca.